File: ican/frontend/src/api/documents.js
import { api } from './client'

export function listDocuments() {
  return api.get('/documents').then((r) => r.data)
}

export function uploadDocument(file, scene) {
  const form = new FormData()
  form.append('file', file)
  form.append('scene', scene)
  return api.post('/documents/upload', form).then((r) => r.data)
}

export function deleteDocument(id) {
  return api.delete(`/documents/${id}`).then((r) => r.data)
}

export function listSessions() {
  return api.get('/sessions').then((r) => r.data)
}

export function createSession() {
  return api.post('/sessions').then((r) => r.data)
}

export function getSession(id) {
  return api.get(`/sessions/${id}`).then((r) => r.data)
}

export function deleteSession(id) {
  return api.delete(`/sessions/${id}`).then((r) => r.data)
}

/** 在线校验用户自带的模型 Key 是否可用 */
export function verifyKey(provider, apiKey, model = '') {
  return api.post('/config/verify-key', { provider, api_key: apiKey, model }).then((r) => r.data)
}

File: ican/backend/app/api/documents.py
"""文档管理接口：上传 / 列表 / 删除。"""
import logging

from fastapi import APIRouter, File, Form, HTTPException, UploadFile

from app.schemas import DocumentOut
from app.services import doc_service

logger = logging.getLogger(__name__)
router = APIRouter(tags=["documents"])

_ALLOWED_SCENES = {"study", "research", "career", "campus"}


@router.post("/documents/upload")
async def upload_document(
    file: UploadFile = File(...),
    scene: str = Form("study"),
) -> DocumentOut:
    if scene not in _ALLOWED_SCENES:
        raise HTTPException(status_code=400, detail=f"scene 仅支持 {sorted(_ALLOWED_SCENES)}")
    filename = file.filename or "未命名.pdf"
    if not filename.lower().endswith(".pdf"):
        raise HTTPException(status_code=400, detail="目前仅支持 PDF 文件")
    data = await file.read()
    try:
        result = doc_service.ingest_pdf(filename, data, scene=scene)
    except ValueError as e:
        raise HTTPException(status_code=400, detail=str(e))
    except Exception:
        logger.exception("文档解析失败: %s", filename)
        raise HTTPException(status_code=500, detail="文档解析失败，请重试")
    return DocumentOut(builtin=False, **result)


@router.get("/documents")
async def list_documents() -> list[DocumentOut]:
    return [
        DocumentOut(**{**d, "builtin": bool(d["builtin"])})
        for d in doc_service.list_documents()
    ]


@router.delete("/documents/{doc_id}")
async def delete_document(doc_id: str) -> dict:
    ok = doc_service.delete_document(doc_id)
    if not ok:
        raise HTTPException(status_code=404, detail="文档不存在")
    return {"ok": True}
